apps/infrastructure/cli/cli.py

```diff
@@ -50,65 +72,78 @@ def cli(config, output_file):
-
-    ## Get app config and arguments
-    config.app = Config(name=app.lower())
-
-    ## Deployment type
-    config.deployment_type = (
-        "serverless"
-        if click.confirm(f"Do you want to deploy serverless?")
-        else "serverfull"
-    )
-
-    ## Websockets
-    config.websockets = (
-        True if click.confirm(f"Will you need to support Websockets?") else False
-    )
-
-    get_app_arguments(config)
-
-    ## Prompting user to provide configuration for the selected cloud
-    if config.provider == "aws":
-        config.vpc = aws.get_vpc_config()
-    elif config.provider == "gcp":
-        pass
-    elif config.provider == "azure":
-        pass
+    # prev_config = None  # Comment this for dev
 
-    ## Database
-    credentials.db = aws.get_db_config()
+    if prev_config is not None:
+        with open(prev_config, "r") as f:
+            click.echo("loading previous configurations...")
+            config.update(**json.load(f))
+    else:
+        config.provider = provider.lower()
+
+        # Store credentials in a separate object, thus not logging it in output
+        # when asking the user to confirm the current configuration
+        credentials = Config()
+
+        ## credentials file
+        with open(
+            click.prompt(
+                f"Please enter path to your  {colored(f'{config.provider} credentials')} json file",
+                type=str,
+                default=f"{Path.home()}/.{config.provider}/credentials.json",
+            ),
+            "r",
+        ) as f:
+            credentials.cloud = json.load(f)
+
+        ## Get app config and arguments
+        config.app = Config(name=app.lower())
+
+        ## Deployment type
+        config.serverless = False
+        if not config.app.name in ["node", "worker"]:
+            config.serverless = click.confirm(f"Do you want to deploy serverless?")
+
+        ## Websockets
+        if not config.serverless:
+            config.websockets = click.confirm(f"Will you need to support Websockets?")
+
+        get_app_arguments(config)
+
+        ## Prompting user to provide configuration for the selected cloud
+        if config.provider == "aws":
+            config.vpc = aws.get_vpc_config()
+            if not config.serverless:
+                config.vpc.instance_type = aws.get_instance_type()
+        elif config.provider == "gcp":
+            pass
+        elif config.provider == "azure":
+            pass
+
+        ## Database
+        credentials.db = aws.get_db_config()
 
     if click.confirm(
-        f"""Your current configration are: \n\n{colored((json.dumps(vars(config), indent=2, default=lambda o: o.__dict__)))} \n\nContinue?"""
+        f"""Your current configration are:
+        \n\n{colored((json.dumps(vars(config),
+                        indent=2, default=lambda o: o.__dict__)))}
+        \n\nContinue?"""
     ):
 
+        credentials = config.credentials  # Uncomment this for dev
```

Comment:
`credentials = config.credentials  # Uncomment this for dev` Needs to be commented out. It's added only for dev purpose. We would remove this later.

Reply:
Done.